compiler/backend/ccgstmts.nim

```diff
@@ -16,13 +16,11 @@ const
     # above X strings a hash-switch for strings is generated
 
 proc isAssignedImmediately(conf: ConfigRef; n: CgNode): bool {.inline.} =
-  if n.kind == cnkEmpty: return false
-  if isInvalidReturnType(conf, n.typ):
-    # var v = f()
-    # is transformed into: var v;  f(addr v)
-    # where 'f' **does not** initialize the result!
-    return false
-  result = true
+  # note: the destination can be considered assigned immediately even if the
+  # source is an exception-raising RVO call. This is because the call happens
+  # *before* the definition, meaning that nothing on the exceptional control-
+  # flow path must observe (read or write) the destination
```

Comment:
Here "exceptional control flow path" is exclusive of both prior to an exception or in a finally? Basically, one of those latter two needs to write to the destination in order for it to be initialized?